sdk/identity/azure-identity/azure/identity/_internal/client_credential_base.py

```diff
@@ -20,6 +20,7 @@ class ClientCredentialBase(MsalCredential, GetTokenMixin):
     def _acquire_token_silently(self, *scopes: str, **kwargs: Any) -> Optional[AccessToken]:
         app = self._get_app(**kwargs)
         request_time = int(time.time())
+        kwargs.pop("enable_cae", None)
         result = app.acquire_token_silent_with_error(
```

Comment:
Instead of popping, let's specify specific kwargs when calling into msal app so there will not be arbitrary args.

Reply:
Yea, I can try to construct an allow-list for kwargs we can verify against. Ultimately, these kwargs are passed to requests which accepts a number of [keyword args](https://requests.readthedocs.io/en/latest/api/#requests.request).

Since none of our other calls to MSAL methods pass through kwargs, seems that this kwargs pass-through may have been extraneous, and it's unclear if it's even needed. How inclusive do you think the list of acceptable kwargs should be (i.e. should I just check for top level keyword arguments that `acquire_token_silent_with_error` takes)?

Reply:
@xiangyan99 Updated to check a list of known kwargs. Hopefully sufficient?